announcement/models/res_users.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    pendant_announcement_ids = fields.Many2many(
+        comodel_name="announcement", relation="pendant_announcement_user_rel",
+    )
+    read_announcement_ids = fields.Many2many(
+        comodel_name="announcement", relation="read_announcement_user_rel",
+    )
+    popup_announcements = fields.Boolean()
+
+    @api.model
+    def announcement_user_count(self):
+        """The js widget gathers the announcements from this method"""
+        now = fields.Datetime.now()
+        group_announcements = self.env["announcement"].search_read(
+            [
+                ("announcement_type", "=", "user_group"),
+                "|",
+                ("notification_date", "<=", now),
+                ("notification_date", "=", False),
+                "|",
+                ("notification_expiry_date", ">=", now),
+                ("notification_expiry_date", "=", False),
+                ("id", "not in", self.env.user.read_announcement_ids.ids),
+            ],
+            ["user_group_ids"],
+        )
+        announcements = self.env["announcement"].browse(
+            {
+                x["id"]
+                for x in group_announcements
+                if any(
+                    [g for g in x["user_group_ids"] if g in self.env.user.groups_id.ids]
+                )
+            }
+        )
+        announcements |= self.env.user.pendant_announcement_ids.filtered(
```

Comment:
Why this OR ? Can't be done with the initial search?

Reply:
It's not an `OR`. We're joining the two types of announcements in a single recordset of unique records.

Reply:
And this is can't be done with a bigger domain?

Reply:
I want to avoid the heavy field `pendant_announcement_ids` + later `filtered`

Reply:
But pendant announcement_ids will only contain a handful of records at the very best. It's not going to be that big.

Reply:
OK then.